### DISCUSSION TITLE
Trying to create an endpoint with a 'generic' parameter

### DISCUSSION THREAD

**Discussion by ShawnClake** [fastapi/fastapi#8158] [Questions] [answered]
**Description**

I'm trying to build an API using FastAPI that needs to integrate with another (undocumented) system which I do not have control over. For this reason, I need to be able to create an endpoint that I can use as an integration point without knowing what POST body data will be sent to my API. Is this possible using FastAPI? Am I able to create a sort of 'generic' parameter for an endpoint function which will capture any POST body data into a dictionary?

Thanks in advance. 

To clarify:
This is simply so that I can reverse engineer the system I don't have control over. Once I see what data is being sent, I can rewrite the endpoints to be more explicit.

**Comment by tiangolo** [comment, marked as answer]
Yep, if you know it's a dict, declare as `dict`.

If it might be a `list`, then use `Any`.

---

I received an email notification with the same answer by @dmontagu, but I don't see it here... 🤷‍♂️

**Comment by dmontagu** [comment]
@tiangolo I deleted because I wasn’t sure if it would work if you want a dict with arbitrary keys at the *top* level, and I couldn’t test it out right now (didn’t want to leave unhelpful advice if I was wrong). I figured if you annotate with `x: dict`, for example, it would need to have the key `x` (and the *value* for that key would be the arbitrary dict), but reviewing the docs it seems like annotating a body parameter as dict should just work and get the whole body. Sorry for any confusion!

**Comment by ShawnClake** [comment]
Thank you for the quick answers! I appreciate it.

**Comment by tiangolo** [comment]
Thanks for the help @dmontagu ! :taco: :cake: 

And thanks @ShawnClake for reporting back and closing the issue :tada:
